akka-http-core/src/main/scala/akka/http/impl/engine/rendering/RenderSupport.scala

```diff
@@ -79,13 +79,13 @@ private object RenderSupport {
     override def onPush(elem: ByteString, ctx: Context[ByteString]): SyncDirective = {
       sent += elem.length
       if (sent > length)
-        throw InvalidContentLengthException(s"HTTP message had declared Content-Length $length but entity data stream amounts to more bytes")
+        ctx fail InvalidContentLengthException(s"HTTP message had declared Content-Length $length but entity data stream amounts to more bytes")
       ctx.push(elem)
     }
 
     override def onUpstreamFinish(ctx: Context[ByteString]): TerminationDirective = {
       if (sent < length)
-        throw InvalidContentLengthException(s"HTTP message had declared Content-Length $length but entity data stream amounts to ${length - sent} bytes less")
+        ctx fail InvalidContentLengthException(s"HTTP message had declared Content-Length $length but entity data stream amounts to ${length - sent} bytes less")
```

Comment:
is there a reason for using infix notation here when all other method calls on `ctx` are "normal" ones?